### PR TITLE
Properly account for self ty in method disambiguation suggestion

### LINKED ISSUE #116703: Suggested fix doesn't work in error [E0034]
### Code

```Rust
trait A {
    fn foo(&self);
}

trait B {
    fn foo(&self);
}

struct S;

impl A for S {
    fn foo(&self) {}
}

impl B for S {
    fn foo(&self) {}
}

fn main() {
    let s = S;
    S::foo(&s);
}
```

### Current output

```Shell
help: disambiguate the method for candidate #1
   |
21 |     <&S as A>::foo(&s);
   |     ~~~~~~~~~~~
```

### Desired output

```Shell
help: disambiguate the method for candidate #1
   |
21 |     <S as A>::foo(&s);
   |     ~~~~~~~~~~~
```

### Rationale and extra context

Current output has an extra `&` in the suggested fix that fails compilation.

### Other cases

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by estebank**
Fix #116703.

### CODE REVIEW

File: compiler/rustc_hir_typeck/src/method/suggest.rs [hunk 1]
```diff
@@ -1365,6 +1366,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                             err,
                             path,
                             rcvr_ty,
+                            rcvr_ty,
```
Comment (compiler-errors) [inline compiler/rustc_hir_typeck/src/method/suggest.rs:1369]: Why is this correct? It doesn't seem correct.

File: compiler/rustc_hir_typeck/src/method/suggest.rs [hunk 2]
```diff
@@ -3107,6 +3109,7 @@ fn print_disambiguation_help<'tcx>(
     err: &mut Diagnostic,
     trait_name: String,
     rcvr_ty: Ty<'_>,
+    self_ty: Ty<'_>,
```
Comment (compiler-errors) [inline compiler/rustc_hir_typeck/src/method/suggest.rs:3112]: This is the self type of the method's impl? It should have a better name, imo.

Comment (estebank) [reply compiler/rustc_hir_typeck/src/method/suggest.rs:3112]: What would be a better name? `impl_self_ty` or something?

Comment (compiler-errors) [reply compiler/rustc_hir_typeck/src/method/suggest.rs:3112]: Yeah that works

**Comment by compiler-errors** [comment]
r=me

**Comment by estebank** [comment]
@compiler-errors added a slight rewording to existing suggestions so that all related suggestions use the same text.

@bors r=compiler-errors

**Pull request merged**